#133 code parser alternative (#146)
insightsengineering/teal.code: NEWS.md

# teal.code 0.4.1.9000

# New features

> 

Author: I would go with: 
> * Code lines passed to `new_qenv` and `eval_code` are now assigned to respective objects that they build, so code reproducibility is improved

> 

Reviewer: ok

> 

Author: d1023174911e973af19674b82ebcad5ac871192b

> * `qenv` object now returns `@code` field as `character` (previously it was `expression`)

Reviewer 2: ```suggestion
* `@code` slot in `qenv` object is now a `character` (previously it was `expression`)
```

> * `qenv` object now returns `@code` field as `character` (previously it was `expression`)

Reviewer: ```suggestion
* The `@code` field in the `qenv` class now holds `character` rather than `expression`.
```

Author: 55428339f9764f128bedd2625b002eba694506eb

specific object

> * you can now specify `# @effect object_name` comment tag at the end of the line for the `character` code input in
> `new_qenv()` and `eval_code()` to specify lines having side-effects on objects, so that they are also returned in 
> `get_code()` (when `names` is used)

Reviewer 2: ```suggestion
* Introduced `# @effect` comment-tag in `code` argument of `eval_code` and `new_qenv` functions to support code relationship detection. See more in functions documentation.
```
